.evergreen/mongodl.py

```diff
@@ -317,6 +318,25 @@ def mdb_version_rapid(version: str) -> bool:
     return tup[1] > 0
 
 
+class Retrier:
+    """Class that handles retry logic.  It performs exponential backoff with a maximum delay of 10 minutes between retry attempts."""
+
+    def __init__(self, retries: int) -> None:
+        self.retries = min(retries, 10)
+        self.attempt = 0
+        assert self.retries > 0
+
+    def retry(self) -> bool:
+        if self.attempt == self.retries:
+            return False
+        LOGGER.warning(
+            f"Download attempt failed, retrying attempt {self.attempt} of {self.retries}"
```

Comment:
This warning message may no longer be accurate given the generality of `Retrier` (as implemented, not specific to download routines). Suggest either renaming the class to `DownloadRetrier` or removing the download-specific detail in the message.

Reply:
Updated the name